<launch>
  <arg name="crop_minx" default="-1.5"/>
  <arg name="crop_maxx" default="1.5"/>
  <arg name="crop_miny" default="-2.0"/>
  <arg name="crop_maxy" default="0"/>
  <arg name="crop_minz" default="0"/>
  <arg name="crop_maxz" default="5"/>
  <arg name="surfel_id" default="0" />

  <!-- Start a nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="pcl_manager_$(arg surfel_id)" args="manager" output="log" />
  <!-- Crop x bounds-->
  <node pkg="nodelet" type="nodelet" name="passthrough_x" args="load pcl/PassThrough pcl_manager_$(arg surfel_id)" output="log">
    <remap from="~input" to="points2"/>
    <rosparam>
      filter_field_name: x
      filter_limit_min: -1.5
      filter_limit_max: 1.5
    </rosparam>
  </node>
  <!-- Crop y bounds-->
  <node pkg="nodelet" type="nodelet" name="passthrough_y" args="load pcl/PassThrough pcl_manager_$(arg surfel_id)" output="log">
    <remap from="~input" to="/passthrough_x/output"/>
    <rosparam>
      filter_field_name: y
      filter_limit_min: -1.5
      filter_limit_max: 1.5
    </rosparam>
  </node>
  <!-- Crop z bounds -->
  <node pkg="nodelet" type="nodelet" name="passthrough_z" args="load pcl/PassThrough pcl_manager_$(arg surfel_id)" output="log">
    <remap from="~input" to="/passthrough_y/output"/>
    <rosparam>
      filter_field_name: z
      filter_limit_min: 0.0
      filter_limit_max: 5.0
    </rosparam>
  </node>

  <!-- Compute the convex hull -->
  <!-- <node pkg="nodelet" type="nodelet" name="convex_hull_$(arg surfel_id)" args="load pcl/ConvexHull2D pcl_manager_$(arg surfel_id)" output="log">
    <remap from="~input"   to="/plane_segmenter/kisalus_filtered_plane_output" />
  </node> -->
  <!-- Extract slope-->
  <!-- <node pkg="surface_normals" type="main.py" name="plane_fit_$(arg surfel_id)" output="screen">
    <param name="id" value="$(arg surfel_id)"/>
    <param name="polygon_topic" value="/convex_hull_$(arg surfel_id)/output_polygon"/>
  </node> -->
</launch>